mne/preprocessing/tests/test_realign.py

```diff
@@ -45,45 +50,85 @@ def test_realign(ratio_other, start_raw, start_other, stop_raw, stop_other):
+    info_other = create_info(["other_stim", "other_signal"], sfreq, ["stim", "eeg"])
+    raw = RawArray(data_raw, info_raw, first_samp=111)  # first_samp shouldn't matter
     other = RawArray(data_other, info_other, first_samp=222)
+    raw.set_meas_date((0, 0))  # meas_date shouldn't matter
+    other.set_meas_date((100, 0))
 
-    # naive processing
-    evoked_raw, events_raw, _, events_other = _assert_similarity(raw, other, n_events)
-    if start_raw == start_other:  # can just naively crop
-        a, b = data_raw[0], data_other[0]
-        n = min(len(a), len(b))
-        corr = np.corrcoef(a[:n], b[:n])[0, 1]
-        min_, max_ = (0.99999, 1.0) if sfreq_raw == sfreq_other else (0.8, 0.9)
-        assert min_ <= corr <= max_
+    # find events and do basic checks
+    evoked_raw, events_raw, _, events_other = _assert_similarity(
+        raw, other, n_events, ratio_other
+    )
+
+    # construct annotations
+    onsets_raw = (events_raw[:, 0] - raw.first_samp) / raw.info["sfreq"]
+    dur_raw = [box_len] * len(onsets_raw)
+    desc_raw = ["raw_box"] * len(onsets_raw)
+    annot_raw = Annotations(onsets_raw, dur_raw, desc_raw)
+    raw.set_annotations(annot_raw)
+
+    onsets_other = (events_other[:, 0] - other.first_samp) / other.info["sfreq"]
+    dur_other = [box_len * ratio_other] * len(onsets_other)
+    desc_other = ["other_box"] * len(onsets_other)
+    annot_other = Annotations(onsets_other, dur_other, desc_other)
+    other.set_annotations(annot_other)
+
+    # onsets/offsets correspond to 0/1 transition in boxcar signals
+    _assert_boxcar_annot_similarity(raw, other)
 
     # realign
-    t_raw = (events_raw[:, 0] - raw.first_samp) / other.info["sfreq"]
+    t_raw = (events_raw[:, 0] - raw.first_samp) / raw.info["sfreq"]
     t_other = (events_other[:, 0] - other.first_samp) / other.info["sfreq"]
     assert duration - 10 <= len(events_raw) < duration
     raw_orig, other_orig = raw.copy(), other.copy()
     realign_raw(raw, other, t_raw, t_other)
 
-    # old events should still work for raw and produce the same result
-    evoked_raw_2, _, _, _ = _assert_similarity(
-        raw, other, n_events, events_raw=events_raw
+    # old events should still work for raw and produce the same evoked data
+    evoked_raw_2, events_raw, _, events_other = _assert_similarity(
+        raw, other, n_events, ratio_other, events_raw=events_raw
     )
     assert_allclose(evoked_raw.data, evoked_raw_2.data)
     assert_allclose(raw.times, other.times)
+
     # raw data now aligned
-    corr = np.corrcoef(raw.get_data([0])[0], other.get_data([0])[0])[0, 1]
-    assert 0.99 < corr <= 1.0
+    corr = np.corrcoef(raw.get_data("data"), other.get_data("data"))
+    assert 0.99 < corr[0, 1] <= 1.0
+
+    # onsets derived from stim and annotations are the same
+    assert_allclose(
+        raw.annotations.onset, events_raw[:, 0] / raw.info["sfreq"], atol=2 / sfreq
+    )
```

Comment:
this `atol` gets used a lot so it might be better to do
```suggestion
    atol = 2 / sfreq
    assert_allclose(
        raw.annotations.onset, events_raw[:, 0] / raw.info["sfreq"], atol=atol
    )
```
and use `atol=atol` below as well